### PR TITLE
fix: setView wasn't navigating correctly, nor was getViewReference getting the right ref

### PR DESCRIPTION
**Pull request by wayfarer3130**
### Context

Some of the view reference/presentation functions were not working correctly, causing differences in navigation and view capabilities.

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

- [] I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: packages/tools/examples/viewReferencePresentation/index.ts [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+      })
+    ) {
+      destViewport.setView(viewRef, viewPres);
+      destViewport.render();
+    } else if (applyPresentation) {
+      destViewport.setView(null, viewPres);
+      destViewport.render();
+    }
+  }
+}
```
Comment (sedghi) [inline packages/tools/examples/viewReferencePresentation/index.ts:362]: I thought you had a synchronizer. I'm surprised to see this issue here; perhaps we need a `PresentationSynchronizer`.

File: packages/tools/examples/viewReferencePresentation/index.ts [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+content.append(instructions);
+
+const leftTopDisplayArea = {
+  storeAsInitialCamera: true,
+  imageArea: [1.1, 1.1],
+  imageCanvasPoint: {
+    imagePoint: [0, 0],
+    canvasPoint: [0, 0],
+  },
+};
```
Comment (sedghi) [inline packages/tools/examples/viewReferencePresentation/index.ts:119]: I noticed we lack documentation for `displayArea` here. Could you please add it? Specifically for `storeAsInitialCamera`, please provide at least a brief explanation about its behavior when set to `false` or `true` (not an entire paragraph, but a concise description).

packages/core/src/types/displayArea.ts

```ts
type DisplayArea = {
  type?: 'SCALE' | 'FIT';
  scale?: number;
  interpolationType?: InterpolationType;
  imageArea?: [number, number]; // areaX, areaY
  imageCanvasPoint?: {
    imagePoint: [number, number]; // imageX, imageY
    canvasPoint?: [number, number]; // canvasX, canvasY
  };
  storeAsInitialCamera?: boolean;
};
```

File: utils/ExampleRunner/example-info.json [hunk 3]
```diff
@@ -138,6 +138,10 @@
         "name": "Resize viewport and change aspect ratio",
         "description": "Resize the viewport and allow various aspect ratios/conditions"
       },
+      "viewReferencePresentation": {
```
Comment (sedghi) [inline utils/ExampleRunner/example-info.json:141]: i think the sculptor tool is also missing in this json file, we need to add that as well

File: packages/docs/docs/concepts/cornerstone-core/viewportReferencePresentation.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+  - Uses `isReferenceCompatible` to determine which of a set of viewports is best suited to navigating to an image
+  - Uses `setView(viewRef)` to navigate to the specified image
+- Restoring an earlier view or converting from stack to volume or vice-versa
+  - Uses `ViewReference` and `ViewPresentation` to store the image information
+- Applying interpolation to image sets
+  - Uses `getViewReference` with a specific image position to get references to
+    images in between or related to nearby annotations to interpolate
+- Resizing and sync display presentation
+  - Uses `getViewPresentation` to get old presentation information and then
+    restores with `setView(null,viewPres)`
```
Comment (sedghi) [inline packages/docs/docs/concepts/cornerstone-core/viewportReferencePresentation.md:29]: Can we change `setView` to an object-based approach, like `setView({})`? It seems too early, and I don't think people use it at all. So, can we address this issue of having to pass `null` as a parameter?

File: packages/core/src/RenderingEngine/BaseVolumeViewport.ts [hunk 5]
```diff
@@ -635,6 +664,128 @@ abstract class BaseVolumeViewport extends Viewport implements IVolumeViewport {
...
+      // Handle same frame of reference navigation
+
+      if (refViewPlaneNormal && !isNegativeNormal && !isSameNormal) {
+        // Need to update the orientation vectors correctly for this case
+        console.log(
+          '********* Changing view plane normal',
+          this.id,
+          refViewPlaneNormal,
+          viewUp
+        );
```
Comment (sedghi) [inline packages/core/src/RenderingEngine/BaseVolumeViewport.ts:759]: remove

**Review by sedghi — COMMENTED** [review]
see my comments please

File: packages/tools/examples/viewReferencePresentation/index.ts [hunk 6]
```diff
@@ -0,0 +1,517 @@
...
+  element.onclick = (evt) => {
+    const clickViewport = getEnabledElement(element)?.viewport;
+    viewport = clickViewport;
+    return false;
+  };
+});
+
+content.appendChild(viewportGrid);
+
+const instructions = document.createElement('p');
```
[the remark this answers is not in the document]
Comment (wayfarer3130) [reply packages/tools/examples/viewReferencePresentation/index.ts:99]: Fixed reset camera
Fixed rotation - note it drives the update only, doesn't reflect changing viewports
Apply orientation removed
Apply presentation renamed to "Presentation to All" (to mean that it applies the presentation to all)
Use green for stack and blue for volume, with different shades to distinguish
Removed store display area.

File: packages/tools/examples/viewReferencePresentation/index.ts (hunk 1 above)
Comment (wayfarer3130) [reply packages/tools/examples/viewReferencePresentation/index.ts:362]: Added notes explaining what this is doing.

File: utils/ExampleRunner/example-info.json (hunk 3 above)
Comment (wayfarer3130) [reply utils/ExampleRunner/example-info.json:141]: That is added.

File: packages/docs/docs/concepts/cornerstone-core/viewportReferencePresentation.md (hunk 4 above)
Comment (wayfarer3130) [reply packages/docs/docs/concepts/cornerstone-core/viewportReferencePresentation.md]: Changed to setViewReference and setViewPresentation as agreed.

File: packages/docs/docs/concepts/cornerstone-core/viewportReferencePresentation.md [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+
+- Remembering how an image is presented to allow displaying the same presentation later,
+  e.g., when a viewport is used to display another stack and then is returned to
+  the original stack.
+- Syncing similar but not identical viewports, for example, syncing some or all presentation
+  attributes between different CT views.
+- Resizing of viewports, used to remember the relative positions so that the
+  image remains in the same "relative" position.
+
+## `setView`
```
Comment (sedghi) [inline packages/docs/docs/concepts/cornerstone-core/viewportReferencePresentation.md:156]: i think this is old api right?

File: packages/core/src/RenderingEngine/Viewport.ts [hunk 8]
```diff
@@ -1697,19 +1707,36 @@ class Viewport implements IViewport {
...
-        this.setDisplayArea(displayArea);
-      }
-      this.setZoom(zoom);
-      if (pan) {
-        this.setPan(vec2.scale([0, 0], pan, zoom) as Point2);
-      }
-      if (rotation >= 0) {
-        this.setRotation(rotation);
-      }
+  public setView(viewRef: ViewReference, viewPres: ViewPresentation) {
```
Comment (sedghi) [inline packages/core/src/RenderingEngine/Viewport.ts:1710]: I thought we removed this no? 
I'm fine with calling those separately, if you are too remove this extra method

**Review by sedghi — COMMENTED** [review]
just want to make sure if we decided to remove the setView in favor of those two or not

**Review by sedghi — APPROVED** [review]

**Pull request merged**